### PR TITLE
feat(replay): Read the has_viewed field on replay records and reflect that in the list

### PR DESCRIPTION
**Pull request by ryan953**
Depends on backend support for the new `has_viewed` field.

Follows the blueprint in https://github.com/getsentry/sentry/blob/master/src/sentry/replays/blueprints/api.md
Relates to https://github.com/getsentry/team-replay/issues/19
Relates to https://github.com/getsentry/sentry/issues/64924

Depends on https://github.com/getsentry/sentry/pull/68628

### CODE REVIEW

File: static/app/views/replays/types.tsx [hunk 1]
```diff
@@ -120,6 +124,7 @@ export const REPLAY_LIST_FIELDS = [
   'count_rage_clicks',
   'duration',
   'finished_at',
+  // 'has_viewed', // TODO(replay): Ask for this field when the backend starts to support it
```
Comment (ryan953) [inline static/app/views/replays/types.tsx:127]:
```suggestion
  'has_viewed',
```

File: static/app/views/replays/replayTable/tableCell.tsx [hunk 2]
```diff
@@ -374,21 +374,25 @@ export function ReplayCell({
...
-            </MainLink>
-          )
-        }
-        user={getUserBadgeUser(replay)}
-        // this is the subheading for the avatar, so displayEmail in this case is a misnomer
-        displayEmail={subText}
-      />
+      <Row gap={1}>
+        <UserAvatar user={getUserBadgeUser(replay)} size={24} />
+        <SubText>
```
Comment (billyvg) [inline static/app/views/replays/replayTable/tableCell.tsx:379]: Does this look different than using `UserBadge`?

File: static/app/views/replays/types.tsx [hunk 3]
```diff
@@ -40,6 +40,10 @@ export type ReplayRecord = {
    * The **latest** timestamp received as determined by the SDK.
    */
   finished_at: Date;
+  /**
+   * Whether the currently authenticated user has seen this replay or not.
+   */
+  has_seen: boolean;
```
Comment (ryan953) [inline static/app/views/replays/types.tsx:46]:
```suggestion
  has_viewed: boolean;
```

File: static/app/views/replays/replayTable/tableCell.tsx (hunk 2 above)
Comment (michellewzhang) [reply static/app/views/replays/replayTable/tableCell.tsx:379]: it's looking the same to me:

SCR-20240411-ivmn

Comment (ryan953) [reply static/app/views/replays/replayTable/tableCell.tsx:379]: i'll followup soon with some stories to differentiate the avatar stuff. There's too many components related to that :(

**Review by billyvg — APPROVED** [review]

**Pull request merged**